### PR TITLE
build: remove console.log from the code

### PR DESCRIPTION
**Pull request by EvanBurchard**
Removes the console from the code via the
babel plugin. Additionally the eslint rule
for no console was removed.

Fixes issue [1669](https://github.com/opencollective/opencollective/issues/1669)

### CODE REVIEW

**Comment by EvanBurchard** [comment]
@Betree @znarf One interesting consequence of this is that after the conosle.log linting errors are hidden, the warnings seem better and the linter is useful again.

```

/Users/evanburchard/projects/opencollective/frontend/components/CreateCollectiveForm.js
  208:9  warning  Do not mutate state directly. Use setState()  react/no-direct-mutation-state

/Users/evanburchard/projects/opencollective/frontend/components/EditTwitterAccount.js
  107:7  warning  Do not mutate state directly. Use setState()  react/no-direct-mutation-state
  111:9  warning  Do not mutate state directly. Use setState()  react/no-direct-mutation-state

/Users/evanburchard/projects/opencollective/frontend/components/collective-page/hero/HeroTotalCollectiveContributionsWithData.js
  18:9  warning  The field CollectiveStatsType.totalAmountRaised is deprecated. 2019-08-22: Referals are not supported anymore  graphql/no-deprecated-fields

/Users/evanburchard/projects/opencollective/frontend/lib/graphql/queries.js
  386:9  warning  The field CollectiveStatsType.totalAmountRaised is deprecated. 2019-08-22: Referals are not supported anymore  graphql/no-deprecated-fields
  713:9  warning  The field CollectiveStatsType.totalAmountRaised is deprecated. 2019-08-22: Referals are not supported anymore  graphql/no-deprecated-fields

/Users/evanburchard/projects/opencollective/frontend/lib/stripe.js
  15:7  warning  Possible race condition: `stripeInstances[instanceId]` might be reassigned based on an outdated value of `stripeInstances[instanceId]`  require-atomic-updates

✖ 7 problems (0 errors, 7 warnings)
```

File: .babelrc [hunk 1]
```diff
@@ -14,6 +14,7 @@
   ],
   "plugins": [
     "babel-plugin-styled-components",
+    ["transform-remove-console", { "exclude": [ "error"] }],
```
Comment (Betree) [inline .babelrc]: Can we have this only for `production` env?

**Comment by EvanBurchard** [comment]
@Betree sure. updated.

**Comment by EvanBurchard** [comment]
I rebased from master and now am seeing some failures that seem unrelated to my changes. Holding off on investigating for now.

**Comment by maddhruv** [comment]
@EvanBurchard can you check why there are so many `package-lock.json` changes, which are mostly removal of dependencies.

**Comment by EvanBurchard** [comment]
> @EvanBurchard can you check why there are so many `package-lock.json` changes, which are mostly removal of dependencies.

It's strange. The only package change I made was: `"babel-plugin-transform-remove-console": "^6.9.4"`

`npm i` on master only gave me a tiny diff, so I reset and ran `npm i` again. Smaller diff this time. I'm not sure why.

File: .eslintrc.json [hunk 2]
```diff
@@ -3,7 +3,6 @@
   "env": { "jest": true },
   "plugins": ["graphql", "react-hooks"],
   "rules": {
-    "no-console": "warn",
```
Comment (maddhruv) [inline .eslintrc.json]: why exactly have you removed this warning? 🤔

File: .babelrc [hunk 3]
```diff
@@ -28,6 +28,12 @@
     },
     "production": {
       "plugins": [
+        [
+          "transform-remove-console",
+          {
+            "exclude": "error"
```
Comment (maddhruv) [inline .babelrc]:
```suggestion
            "exclude": ["error"]
```

**Comment by maddhruv** [comment]
the `build` was successful in the local
@Betree can you check why the `now` deployment is failing
`Not able to load user` this error doesn't give enough clue to me

**Comment by znarf** [comment]
- I still want to ultimately move forward with the removal of console.log and would like to keep them as warnings.
- If there is `console.log` in scripts are they now removed when using `npx babel-node scripts/*.js`? That should not be the case. Can we just do it for `pages` and `components`?

**Comment by znarf** [comment]
Closing for now. This will be solved differently.

**Pull request closed without merging**